modules/gravatar-hovercards.php

```diff
@@ -149,6 +169,40 @@ function grofiles_get_avatar( $avatar, $author ) {
 				grofiles_gravatars_to_append( $user->ID );
 		}
 	} else if ( isset( $author->comment_type ) ) {
+		if ( $is_amp ) {
+			if ( 1 === preg_match( '/avatar\/([a-zA-Z0-9]+)\?/', $avatar, $email_hash ) ) {
+				$email_hash  = $email_hash[1];
+				$cache_group = 'gravatar_profiles_';
+				$cache_key   = 'gravatar_profile_' . $email_hash;
+
+				$response_body = wp_cache_get( $cache_key, $cache_group );
+				if ( false === $response_body ) {
+					$response = wp_remote_get( 'https://en.gravatar.com/' . $email_hash . '.json' );
```

Comment:
Maybe we can escape the URL we use for the remote request?

Reply:
@jeherve I've added escaping for the URL as well as the vars that come back from the Gravatar endpoint.